moduleframework/mtf_scheduler.py

```diff
@@ -0,0 +1,163 @@
+    args, unknown = parser.parse_known_args()
+
+    set_logging(level=logging.DEBUG if args.verbose else logging.INFO)
+
+    logger.debug("Options: verbose={0}, name={1}, linter={2}, setup={3}, action={4}, module={5}".format(
+        args.verbose, args.name, args.linter, args.setup, args.action, args.module))
+    logger.debug("remaining options for avocado: {0}".format(unknown))
+
+    # environment
+    # hint for read: os.environ.get('MODULE') ; write os.environ['MODULE']
+    if os.environ.get('MODULE') is None:
+        os.environ['MODULE'] = args.module
+        logger.debug("set up environment {}".format(os.environ['MODULE']))
+    logger.debug("MODULE={0}".format(os.environ.get('MODULE')))
+
+
+    return args, unknown
+
+class AvocadoStart(object):
+    def __init__(self, args, unknown):
+
+        self.module = os.environ['MODULE']
+        self.action = args.action
+
+        # its used for filepath in loadCli:
+        self.tmppath = tempfile.mktemp()
+        site_libs = os.path.dirname(moduleframework.__file__)
+        mtf_tools = "tools"
+
+        # choose between TESTS and ADDITIONAL ENVIRONMENT from options
+        self.tests = ''
+        if args.linter:
+            self.tests = ("{SITE_LIB}/{MTF_TOOLS}/*.py".format(SITE_LIB=site_libs, MTF_TOOLS=mtf_tools))
+        self.additionalAvocadoArg = ''
+
+        for param in unknown:
+            # take care of this, see tags for safe/unsafe:
+            # http://avocado-framework.readthedocs.io/en/52.0/WritingTests.html#categorizing-tests
+            if os.path.exists(param):
+                # this is test
+                self.tests += " {0} ".format(param)
+            else:
+                # this is additional avocado param
+                self.additionalAvocadoArg += " {0} ".format(param)
+
+        logger.debug("tests = {0}".format(self.tests))
+        logger.debug("additionalAvocadoArg = {0}".format(self.additionalAvocadoArg))
+
+
+    def bash_command(self, cmd):
+         return process.run(cmd, shell=True, ignore_status=True)
+
+    def avocado_run(self):
+        avocado_args="--json {JSON_LOG}".format(JSON_LOG=self.tmppath)
+        avocadoAction = "MODULE={MODULE} avocado {ACTION} {AVOCADO_ARGS}".format(MODULE=self.module, ACTION=self.action, AVOCADO_ARGS=avocado_args)
+
+        # run avocado with right cmd arguments
+        bash = self.bash_command(("{AVOCADO} {a} {b}".format(AVOCADO=avocadoAction, a=self.additionalAvocadoArg, b=self.tests)))
+        logger.info(bash.stdout)
+        logger.debug("cmd was {0}".format(bash.command))
+
+
+    def show_error(self):
+        # show errors details and delete log file
+        if os.path.exists(self.tmppath):
+            cmd = self.bash_command("mtf-log-parser {JSON_LOG}".format(JSON_LOG=self.tmppath))
+            logger.info(cmd.stdout)
+            logger.debug("cmd was {0}".format(cmd.command))
+            os.remove(self.tmppath)
+
+def main():
+    args, unknown = cli()
+
+    if args.setup:
```

Comment:
Is it plan to remove mtf-env-setup? I don't understant, why it is part of `mtf`

Reply:
I add -s / --setup  to mtf, default is 'false' and yes it calls 'mtf-env-setup'
I feel it could do this setup, or not?